### PR TITLE
Improve the ReadImage and WriteImage procedures.

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

**Review by dave3d — APPROVED** [review]
Looks good

File: Code/IO/include/sitkImageFileWriter.h [hunk 1]
```diff
@@ -165,10 +165,22 @@ class SmartPointer;
...
+  /**
+   * \brief WriteImage is a procedural interface to the ImageFileWriter.
+   *     class which is convenient for many image writing tasks.
+   *
+   *  \param image the input image to be written
+   *  \param fileName the filename of an Image e.g. "cthead.mha"
+   *  \param useCompression request to compress the written file
+   *  \param compressionLevel a hint for the amount of compression to
+   *    be applied during writing
+   *
```
Comment (zivy) [inline Code/IO/include/sitkImageFileWriter.h:177]: Copy the documentation for useCompression and compressionLevel from the [ImageFileWriter](https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1ImageFileWriter.html#aee65575333679cdf0a91a62386752ccb) as those provide more details.

File: Code/IO/include/sitkImageSeriesWriter.h [hunk 2]
```diff
@@ -152,12 +152,26 @@ class ImageIOBase;
...
+   * \brief WriteImage is a procedural interface to the ImageSeriesWriter.
+   *     class which is convenient for many image writing tasks.
+   *
+   *  \param image the input image to be written
+   *  \param fileNames a vector of filenames of length equal to the
+   *  number of slices in the image.
+   *  \param useCompression request to compress the written file
+   *  \param compressionLevel a hint for the amount of compression to
+   *    be applied during writing.
+   *
```
Comment (zivy) [inline Code/IO/include/sitkImageSeriesWriter.h:166]: Same comment as above with respect to documentation of useCompression and compressionLevel

File: Wrapping/Python/SimpleITK/extra.py [hunk 3]
```diff
@@ -298,4 +300,98 @@ def GetImageFromArray(arr, isVector=None):
...
+
+    Parameters
+    ----------
+    fileName
+     A single or a list of file names. the filename of an Image e.g. "cthead.mha"
+    outputPixelType
+     The pixel type of the returned Image. By default the value is sitkUnknown, which enable the output pixel type to
+     be same as the file. If the pixel type is specified then the itk::ConvertPixelBuffer will be used to convert the
+     pixels.
+    imageIO
```
Comment (zivy) [inline Wrapping/Python/SimpleITK/extra.py:320]: "By default the value is sitkUnknown, which enable the output pixel type to be same as the file." ->

"By default the value is sitkUnknown, which indicates that the output pixel type be the pixel type stored in the file."

File: Wrapping/Python/SimpleITK/extra.py [hunk 4]
```diff
@@ -298,4 +300,98 @@ def GetImageFromArray(arr, isVector=None):
...
+               compressionLevel=-1,
+               *,
+               imageIO: str = "",
+               compressor: str = "") -> "None":
+    r"""
+    WriteImage is a procedural interface to the ImageFileWriter and ImageSeriesWriter classes which is convenient for
+    many image writing tasks.
+
+    For an input image of N dimensions, a series of N-1 dimensional (slices) images can be written by providing a list
+    if file names equal to the number of slices in the input image.
```
Comment (zivy) [inline Wrapping/Python/SimpleITK/extra.py:361]: a list if -> a list of

**Review by zivy — APPROVED** [review]
Several minor comments with respect to the documentation.

**Pull request merged**